cmd/iam.go

```diff
@@ -983,7 +983,7 @@ func (sys *IAMSys) NewServiceAccount(ctx context.Context, parentUser string, gro
 
 	var accessKey, secretKey string
 	var err error
-	if len(opts.accessKey) > 0 {
+	if len(opts.accessKey) > 0 || len(opts.secretKey) > 0 {
```

Comment:
auth.CreateNewCredentialsWithMetadata seems to always fail if access key or secret key is empty. I think it is better to add more check so that we require both are empty or both are non empty and we return an error early in that case.